### PR TITLE
Bump ORCA version to 3.83, allow array cmp on btree index when using bitmap scan

### PR DESCRIPTION
**Pull request by zellerh**
The corresponding ORCA PR is https://github.com/greenplum-db/gporca/pull/554.

Change the check when translating an ORCA query to a plan.
The old check prohibited ArrayCmp on a btree index. The new
check is similar, except that it allows an ArrayCmp on a
btree index when it is done in a bitmap index probe.

Updated ICG result files and added a new test case.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/regress/expected/gporca.out [hunk 1]
```diff
@@ -10465,48 +10465,87 @@ drop function canSetTag_Func(x int);
 drop table canSetTag_bug_table;
 drop table canSetTag_input_data;
 -- Test B-Tree index scan with in list
-CREATE TABLE btree_test as SELECT * FROM generate_series(1,100) as a distributed randomly;
+CREATE TABLE btree_test as SELECT i a, i b FROM generate_series(1,100) i distributed randomly;
 CREATE INDEX btree_test_index ON btree_test(a);
+set optimizer_enable_tablescan = off;
+-- start_ignore
+select disable_xform('CXformSelect2IndexGet');
```
Comment (asimrp) [inline src/test/regress/expected/gporca.out]: Basic question, for my own understanding: why is an Orca specific parameter being changed when the explain plan below indicates that PostgreSQL query optimizer is being used?  Is Orca expected to fall back to PostgreSQL planner?

Comment (zellerh) [reply src/test/regress/expected/gporca.out]: This file is run with and without Orca enabled, the Orca results are shown in file gporca_optimizer.out. We shouldn't fall back for these queries.

File: src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp [hunk 2]
```diff
@@ -809,7 +811,9 @@ CTranslatorDXLToPlStmt::TranslateIndexConditions
 		GPOS_ASSERT((IsA(index_cond_expr, OpExpr) || IsA(index_cond_expr, ScalarArrayOpExpr))
 				&& "expected OpExpr or ScalarArrayOpExpr in index qual");
 
-		if (IsA(index_cond_expr, ScalarArrayOpExpr) && IMDIndex::EmdindBitmap != index->IndexType())
+		if (!is_bitmap_index_probe &&
+			IsA(index_cond_expr, ScalarArrayOpExpr) &&
+			IMDIndex::EmdindBitmap != index->IndexType())
 		{
 			GPOS_RAISE(gpdxl::ExmaDXL, gpdxl::ExmiDXL2PlStmtConversion, GPOS_WSZ_LIT("ScalarArrayOpExpr condition on index scan"));
```
Comment (chrishajas) [inline src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp]: I searched for `ScalarArrayOpExpr condition` in the ICW output and found a couple of instances that may be fixed with this change (`gp_gin_index.sql`). Are there any indexes with ScalarArrayOpExpr that we don't support now?

Comment (zellerh) [reply src/backend/gpopt/translate/CTranslatorDXLToPlStmt.cpp]: Sorry, the test cases in gp_gin_index.sql are still failing the same way, this PR should not change behavior with GIN indexes. ORCA should now support all the cases that planner supports (natively, by falling back, or by substituting a bitmap scan for an index scan).

Here is the current behavior for array comparisons (IN predicate with a constant list) as I understand it for btree indexes (no change for other index types):

|  | 5X | 6X | Master |
| --- | --- | --- | --- |
| planner  | bitmap scan | index scan | index scan |
| ORCA before before PR https://github.com/greenplum-db/gporca/pull/553 | fallback | fallback | fallback |
| ORCA after PR https://github.com/greenplum-db/gporca/pull/553 | bitmap or seq scan | bitmap or seq scan | bitmap or seq scan |
| ORCA with this PR | bitmap scan | bitmap scan | bitmap scan |

**Pull request merged**